pslab/instrument/multimeter.py

```diff
@@ -125,48 +125,26 @@ def measure_capacitance(self) -> float:
             Capacitance in Farad.
         """
         for current_range in self._CURRENTS_RANGES:
-            for i, charge_time in enumerate([50000, 5000, 500, 50, 5]):
-                voltage, _ = self._measure_capacitance(current_range, 0, charge_time)
-
-                if voltage < self._CAPACITOR_CHARGED_VOLTAGE:
-                    if i:
-                        return self._binary_search_capacitance(
-                            current_range, charge_time, charge_time * 10
-                        )
-                    else:
-                        break  # Increase current.
+            charge_time = 10
+            for i in range(10):
+                if charge_time > 50000:
+                    break  # Increase current.
+                voltage, capacitance = self._measure_capacitance(
+                    current_range, 0, charge_time
+                )
+                if (
+                    voltage / self._CAPACITOR_CHARGED_VOLTAGE > 0.98
+                    and voltage / self._CAPACITOR_CHARGED_VOLTAGE < 1.02
+                ):
+                    return capacitance
+                elif self._CAPACITOR_CHARGED_VOLTAGE / voltage > 1.1:
+                    charge_time = int(
+                        charge_time * self._CAPACITOR_CHARGED_VOLTAGE / voltage
+                    )
```

Comment:
```suggestion
                charge_time = int(
                    charge_time * self._CAPACITOR_CHARGED_VOLTAGE / voltage
                )
```

I think the charge time update should be unconditional. Otherwise, there are voltage values for which capacitance is not returned and the charge time is not updated; instead, the measurement will be run again with the same parameters, which should yield the same voltage.

Reply:
You are right ! I had initially thought of the same thing.